src/api.rs

```diff
@@ -1,33 +1,151 @@
 // API endpoints
 
-use actix_web::{post, get, Responder, web::Payload};
+use std::fmt::{Display, Formatter, write};
+use std::ops::DerefMut;
+use std::str::FromStr;
+use std::sync::RwLock;
+use actix_web::{post, get, Responder, web, Result, ResponseError, Error};
+use actix_web::http::StatusCode;
+use actix_web::web::Json;
+use serde::{Serialize, Deserialize};
+use serde_json::{json};
+use crate::api::ValidationError::NoSuchGameError;
+use crate::game::{GameManager, adapter, connect4, GameId};
+use crate::game::adapter::GameAdapter;
+
+/* Helpers */
+
+/// ValidationError
+#[derive(Debug, Clone)]
+pub enum ValidationError {
+    ParseIntError(String),
+    NoSuchGameError(String)
+}
+
+impl Display for ValidationError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ValidationError::ParseIntError(id) => write!(f, "ID {} is Invalid", id),
+            ValidationError::NoSuchGameError(game) => write!(f, "Game {} not found", game)
+        }
+    }
+}
+
+impl ResponseError for ValidationError {
+    fn status_code(&self) -> StatusCode {
+        StatusCode::BAD_REQUEST
+    }
+}
+
+/// validate_id
+fn validate_id(game_id: &String) -> Result<u128, Error> {
+    let game_id_int = u128::from_str(&game_id);
+    match game_id_int {
+        Ok(value) => Ok(value),
+        Err(_) => return Err(actix_web::Error::from(ValidationError::ParseIntError(game_id.clone()))) // TODO: need custom error
+    }
+}
+
+/* Structs for the different endpoints*/
+
+/// Struct for req to `/api/create-game`
+#[derive(Deserialize)]
+pub struct CreateGameReq {
```

Comment:
Do these all need to be `pub`? Are they used anywhere else?

Reply:
Ok

Reply:
Yes they do it turns out.